docs/reference/cluster/prevalidate-node-removal.asciidoc

```diff
@@ -0,0 +1,91 @@
+[[prevalidate-node-removal-api]]
+=== Prevalidate node removal API
+++++
+<titleabbrev>Prevalidate node removal</titleabbrev>
+++++
+
+NOTE: {cloud-only}
+
+Prevalidate node removal.
+
+[[prevalidate-node-removal-api-request]]
+==== {api-request-title}
+
+`POST /_internal/prevalidate_node_removal/<nodes>`
+
+[[prevalidate-node-removal-api-prereqs]]
+==== {api-prereq-title}
+
+* If the {es} {security-features} are enabled, you must have the `monitor` or `manage` <<privileges-list-cluster,cluster privilege>> to use this API.
+
+[[prevalidate-node-removal-api-desc]]
+==== {api-description-title}
+
+This API allows validating whether it is safe to remove the specified node(s) from the cluster. In case the cluster has a <<cluster-health,`red` cluster health status>>, this verifies that the removal of the node(s) would not risk removing the last remaining copy of an unassigned shard.
+
+The response includes the overall safety of the removal of the specified nodes, and a detailed response for each node. The node-specific part of the response also includes more details on why removal of that node might not be safe.
+
+The result of the prevalidation is expressed as either `YES`, `NO`, or `UNKNOWN`.
+
+[[prevalidate-node-removal-api-path-params]]
+==== {api-path-parms-title}
+
+`<node>`::
+    (Required, string) Comma-separated list of node IDs, names or external IDs.
+
+[[prevalidate-node-removal-api-query-params]]
+==== {api-query-parms-title}
+
+include::{es-repo-dir}/rest-api/common-parms.asciidoc[tag=master-timeout]
+
+[[prevalidate-node-removal-api-response-body]]
+==== {api-response-body-title}
+
+`result`::
+    (object) Prevalidation result for the removal of all the provided nodes.
+
+`nodes`::
+    (object) Prevalidation result for the removal of each of the provided nodes.
+
+[[prevalidate-node-removal-api-example]]
+==== {api-examples-title}
+
+This example validates whether it is safe to remove the nodes `node1` and `node2`. The response indicates that it is safe to remove `node1`, but it might not be safe to remove `node2`. Therefore, the overall prevalidation of the removal of the two nodes is `UNKNOWN`.
+
+[source,console]
+--------------------------------------------------
+POST /_internal/prevalidate_node_removal/node1,node2
```

Comment:
If we are moving away from having the nodes in the path (I started with that since I thought it would be similar/consistent to some other APIs we have), should we have three query params and allow setting only one of them per request?

```
POST /_internal/prevalidate_node_removal?name=node1,node2
POST /_internal/prevalidate_node_removal?id=id1,id2
POST /_internal/prevalidate_node_removal?external_id=id1,id2
``` 

This avoids any kind of ambiguity and we can just resolve based on id, name or external ID.

I'm not entirely sure how problematic the length of the URL can be. I am assuming up to 2000 is ok and considering ID/names of length 40, and the fact that the number of id/names is in the order of a fraction of the cluster size, I think not using a body should be ok. WDYT?

Reply:
Sounds fine to me.